python/note_input_widget/widget.py

```diff
@@ -360,7 +445,39 @@ def _async_submit_note(self, sg, data):
                                           "tasks": note_tasks })
         
         self.__upload_thumbnail(sg_note_data, sg, data)
+        self.__upload_attachments(sg_note_data, sg, data)
         
+
+    def __upload_attachments(self, parent_entity, sg, data):
+        """
+        Uploads any generic file attachments to Shotgun, parenting
+        them to the Note entity.
+
+        :param parent_entity:   The Note entity to attach the files to in SG.
+        :param sg:              A Shotgun API handle.
+        :param data:            The data dict containing an "attachments" key
+                                housing a list of file paths to attach.
+        """
+        for file_path in data.get("attachments", []):
+            if os.path.exists(file_path):
```

Comment:
Should there at least be a warning if a file we are expecting to upload isn't there?